src/schemas/tokens-info.ts

```diff
@@ -67,6 +69,13 @@ const BaseTokenInfoSchema = Joi.object({
 })
   // Ensure `isFeeCurrency` is present when `isCoreToken` (deprecated) is present.
   .with('isCoreToken', 'isFeeCurrency')
+  // Ensure `isFeeCurrency` is true and `feeCurrencyAdapterDecimals` is present when `feeCurrencyAdapterAddress` is set.
+  .when(Joi.object({ feeCurrencyAdapterAddress: Joi.exist() }).unknown(), {
```

Comment:
nit: Just noting that `.unknown()` is a bit weird, is it needed?

Reply:
yes the syntax is a bit weird, but it's necessary to work.